### DISCUSSION TITLE
Issues  with Blackhole and Jamulus working together

### DISCUSSION THREAD

**Discussion by davidjjj** [ExistentialAudio/BlackHole#613] [Q&A]
I am routing audio from Jamulus to Zoom v.v. through Blackhole. This morning it still worked fine, but from the beginning of the afternoon it no longer works. When I select Blackhole as the input device for Zoom the input volume of Zoom jumps to 0.
I can move the slider up, but as soon as I lift my hand from the mouse button, the slider immediately jumps back to zero.

After trying many times, I decided to do a clean install of my entire MacBook. After that I only installed Jamulus, Zoom and Blackhole. Unfortunately with the same result.

Do you have any idea what the problem can be?

**Comment by devinroth** [comment]
Stuck volume key?

Devin

**Comment by davidjjj** [comment]
I think not. My Macbook has a Touch Bar and no separate volume key. If it is stuck (if possible) than I would assume that it would be solved after a clean install.

I attached a video of what happens.

https://user-images.githubusercontent.com/42859745/110253400-6dd7b680-7f8a-11eb-8a94-a96555b610f3.mov

**Comment by davidjjj** [comment]
I also tried installing a trial version of Loopback. When I switch to loopback, the volume slider does stick to the position I clicked.

**Comment by devinroth** [comment]
Please consider becoming a Comet Sponsor. https://github.com/sponsors/ExistentialAudio

**Comment by devinroth** [comment]
The input volume from BlackHole is disabled. By design. That is not part of your problem. Check BlackHole output volume.

**Comment by davidjjj** [comment]
Thanks Devin for your quick replies!
Meanwhile I was able to test on another Mac, which works as expected (see attached video)

https://user-images.githubusercontent.com/42859745/110301160-0a846d80-7ff8-11eb-9817-ec2223bb94ef.mov

The blackhole output volume on my Macbook is at the maximum level. The only difference is that my MacBook is on Zoom 5.5.4 and the other Mac is on Zoom 5.5.1. The BlackHole versions are identical.

**Comment by davidjjj** [comment]
I just downgraded Zoom on my MacBook to a version below 5.5.1, and the issue persists, so I think I excluded Zoom as the source of the issue.

**Comment by eugenet8k** [comment]
The same thing here in macOS Big Sur v11.12.1 and Zoom v5.4.7, it used to work like a clock, and now it doesn't. Both 2ch and 16ch simply dead silent in Zoom.

**Comment by devinroth** [comment]
I just tested on Big Sur and Zoom and it works fine.

There is probably something else wrong. Like if you’re using a multi-output. 9 times out of 10 some setting in a multi-output got messed up.

Devin

**Comment by davidjjj** [comment]
Hi Devin, thanks for testing. I tried three Macs meanwhile. 2 of them have no issue at all. The third one does, even after a complete reinstall. This reinstall is also why I am absolutely sure I don’t have a multi-output device configured. Furthermore, the issue is with blackhole as Zoom input, not output. @eugenet8k what type of Mac do you have?
I have the issue on a late 2017 MacBook Pro. The devices that work fine are iMacs.

**Comment by devinroth** [comment]
Do you have the volume all the way up on the BlackHole output?

Devin

**Comment by WilliamAhlert** [comment]
I've got the settings for Garageband as follows, and I think the settings weren't like this before, and I couldn't get sound before. Same as in the instructions given: Blachole as the INPUT, and Multi-output device as the output. Then changed settings on zoom so the blackhole is the 'microphone' and my audio interface is the 'speaker'. (USB Audio CODEC). Tested it by having a zoom between my computer and my phone and the audio came through clear, and it wasn't a matter of audio leaking as the computer was sending audio through my headphones, so with the headphones off I was able to hear clear audio of my playing in Garageband through the phone. So, I think that's the solution and it's solved. It worked for me, anyway!

**Comment by WilliamAhlert** [comment]
UPDATE: I've changed it so the Audio Interface is the 'input' on Garageband, and it still works over Zoom. Except now I can directly transmit both my playing and microphone audio from a mic connected to the audio interface over zoom, so now I can do everything that I used to be able to do before the update. I will say I removed the Multi-Output device and made a new one with the Blackhole 16ch and USB Audio CODEC used, with the Blackhole being the master device with drift correction. Hopefully this helps but just thought I would say what worked for me! :)

**Comment by WilliamAhlert** [comment]
Sorry for another update, but your system settings should be different; don't use the multi-ouput device and your system output, just use your audio interface for I/O or whatever it is you wish to hear the audio from. Probably obvious but I made that mistake and made using chrome impossible because of loading issues I think? Changed it so now all is well. Hope this isn't spam and it's actually helpful ahahaha!

**Comment by davidjjj** [comment]
> Do you have the volume all the way up on the BlackHole output? Devin
> […](#)

Yes, in system settings Blackhole output is at max.

**Comment by devinroth** [comment]
Well then explain exactly how you have everything setup with photos.

Hard for me to help you if I can’t reproduce it.

Devin

**Comment by davidjjj** [comment]
Yes I get that. I tested two MacBooks, both don't work and two iMacs that do work. I'll share all settings with you tomorrow. Thx for looking into this.

David

**Comment by eugenet8k** [comment]
@davidjjj I have retested everything again (with reinstall Blackhole with brew), making sure all the output and input volume is maxed for the Blackhole devices. It does not work in Zoom for me. There was a question about what type of Mac I have and it is MacBook Pro (15-inch, 2018).

BTW the setup I use. En external DAW input + Microphone input via Aggregated Device into GarageBand then mixed together and sent to Blackhole Device. Zoom has input set to Blackhole and Output to my headphones. This setup has been working for nearly a year, but things started to be shacky once I upgraded to macOS Big Sur. I can't claim that the OS upgrade broke it as I recall Blackhole worked for me in Big Sur. But just recently (maybe the latest OS upgrade or Zoom) it stopped working. It's just silence. 

I wish there would be a way to access some log info from the Blackhole device to post here? Is there such a thing?

**Comment by devinroth** [comment]
Does it record normally into Audacity?

**Comment by davidjjj** [comment]
Hi Devin, as promised here the detailed info.

I join a Jamulus session. Audio is coming from other users of that session.
In Jamulus I choose Blackhole as output device 
<img width="949" alt="Schermafbeelding 2021-03-10 om 16 47 24" src="https://user-images.githubusercontent.com/42859745/110656541-52ee8780-81c0-11eb-8a3e-ba0ad69783dd.png">

In the audio settings on my Mac blackhole input and output signals are at maximum:
<img width="812" alt="Schermafbeelding 2021-03-10 om 16 48 31" src="https://user-images.githubusercontent.com/42859745/110656720-803b3580-81c0-11eb-9d64-b79bfac5b274.png">
<img width="812" alt="Schermafbeelding 2021-03-10 om 16 48 45" src="https://user-images.githubusercontent.com/42859745/110656737-84675300-81c0-11eb-8b0a-99b08a930aae.png">

In Zoom I select blackhole as the audio device for input and output as I also want to return the audio that comes from Zoom to Jamulus. Here are my Zoom settings:
<img width="846" alt="Schermafbeelding 2021-03-10 om 16 49 59" src="https://user-images.githubusercontent.com/42859745/110657113-dc9e5500-81c0-11eb-8390-b9b810fd18af.png">

I tried sending the audio to Audacity instead of Zoom and selected blackhole as input for Audacity. Also in that case there is no response.
<img width="1552" alt="Schermafbeelding 2021-03-10 om 16 55 15" src="https://user-images.githubusercontent.com/42859745/110657884-81209700-81c1-11eb-83f8-6b13e574065e.png">

Do you want me to repeat those steps on one of the computers that does work?

Thanks, David

**Comment by eugenet8k** [comment]
Hi guys, I take my words back! The Blackhole device works just fine in my env macOS Big Sur. After doing yet another round of verification, I discovered that if I run a recorded track in GarageBand I do hear audio passed to Blackhole in another app using it as Input.

I discovered that in GarageBand you need to turn on Monitor of the Track to hear the output of my mic and/or another device in realtime streamed to Blackhole. I guess that setting was on before, but GarageBand randomly turns it on/off if you change the Audio Device input/ouput in the config. So I guess I didn't notice that little orange button wasn't pressed:

<img width="407" alt="Monitor-Button-GarageBand" src="https://user-images.githubusercontent.com/4804824/110673990-49e0c480-8186-11eb-9bb8-bc6ffeb0810a.png">

**Comment by devinroth** [comment]
Looks like you have BlackHole 2ch set as the input to Jamulus. Not as the output.

Devin

**Comment by devinroth** [comment]
If you play something from YouTube does that go through zoom with this setup?

**Comment by davidjjj** [comment]
> Looks like you have BlackHole 2ch set as the input to Jamulus. Not as the output. Devin
> […](#)

You are right. I am quite sure that this was only during the screen prints, so I just retested and also when I do select blackhole as Jamulus output the results are identical.

**Comment by davidjjj** [comment]
> If you play something from YouTube does that go through zoom with this setup?

Yes! that works...
<img width="831" alt="Schermafbeelding 2021-03-11 om 10 49 01" src="https://user-images.githubusercontent.com/42859745/110768412-816b7180-8257-11eb-9693-3912b6809cff.png">

**Comment by davidjjj** [comment]
When YouTube works it is weird that Jamulus doesn't. And today I do get a signal. I really don't know what is the problem. I cancelled an online concert on Sunday because this did not work anymore (and it did work during the sound check 2 hours before the concert). Also yesterday it did not work yet. And today I am having a signal. Any ideas?
<img width="1091" alt="Schermafbeelding 2021-03-11 om 10 54 41" src="https://user-images.githubusercontent.com/42859745/110769505-87158700-8258-11eb-9e16-9752f22ffdbf.png">

**Comment by devinroth** [comment]
No idea.

Have you asked the Jamulus folks?


Devin

**Comment by LeonardCA** [comment]
> When I select Blackhole as the input device for Zoom the input volume of Zoom jumps to 0.
> I can move the slider up, but as soon as I lift my hand from the mouse button, the slider immediately jumps back to zero.

Starting on 3/9/2021 I am occasionally having the same problem with audio stuck on "mute".  I have been unable to successfully consistently reproduce it.  It has happened 3 times now, and rebooting has fixed the stuck "mute" problem for non-blackhole audio output. 

But, I have also have the problem that within other applications, the multi-output devices including BlackHole no longer work.  It's like BlackHole either stopped taking input, OR BlackHole got its output stuck on "mute" with no way to un-mute it.  I have not been able to fix this problem, and this is mission-critical.

As far as I know, I changed nothing on the day things stopped working (3/9/2021).  I upgraded from MacOS Big Sur 11.1 to 11.2.2 on 2/26/2021.   I upgraded to 11.2.3 on 3/11/2021 in hopes that would fix things, but it did not.  I have tried rebooting to safe mode, then normal mode again to clear any conflicts. 

I use BlackHole with StreamlabsOBS.  I don't think the issue is specific to the application people are using.

**Comment by devinroth** [comment]
Screenshots?

Devin Roth

**Comment by LeonardCA** [comment]
Update: to help others who might have this issue

It looks like somehow in AudioMidi setup the Blackhole 2ch "mute" button did indeed get pressed.  I hadn't opened this application since setting up BlackHole months ago, so I have no idea how this could have happened, but un-checking "mute" for the Blackhole 2ch device fixed the problem.  (Nothing like writing it up publicly to be able to solve the issue...) It still begs the question as to how this happened, but at least I can stream again.
![2021-03-11-AudioMidiBlackHoleMute](https://user-images.githubusercontent.com/80489739/110862560-678c5780-8274-11eb-97da-aeef0451507a.jpg)

**Comment by ghost** [comment]
blackhole 16 on m1 macbook stopped working if you are connected to a monitor and external speakers or airpods2.  If you then select different modes at the exit to the menu, then the recording will go through blackhole 16 only but you will not hear in real time.

**Comment by davidjjj** [comment]
I think I found a way to reproduce the issue today. At first blackhole worked to route audio between Jamulus and Zoom. During setup there were several moments the audio was ringing. After that the audio signal from blackhole was completely gone and there was no way to get it back. I cannot confirm the ringing was the cause, but it was the thing that occurred between a working blackhole and a dead blackhole. Maybe it helps in finding the issue.
To make sure the delayed online concert could still take place today I decided to use loopback instead.

**Comment by chrisrimple** [comment]
Perhaps not the source of any of the problems here, but if you install Loopback (even the free demo) and configure a routing, **that routing remains ON unless you turn it OFF**, and quitting Loopback does NOT turn it OFF.  For example, if you tell Loopback to use the audio from MainStage or Logic as a source, later trying to output from MainStage or Logic to BlackHole will fail because Loopback is intercepting the audio before it can reach BlackHole, and the end result is that Jamulus gets silence as input from BlackHole.
